Add rewrite for scalaz.syntax.std.OptionOps.| syntax

mielientiev opened the pull request.

mielientiev committed:
Add rewrite for scalaz.syntax.std.OptionOps. syntax

gabro reviewed (changes requested):
Thanks for this!
The only minor change I would make is not to introduce a `MigrateOptionSyntax3` but instead put this in `MigrateOptionSyntax` along with the other base cases.

The purpose of `MigrateOptionSyntax2` is to have a Scalaz universal import, but it's not relevant in this case.

Another easy contribution (if you feel like) would be to do the same for the `|` method of `\/`, which becomes `getOrElse` too.

mielientiev commented:
@gabro 
I've removed `MigrateOptionSyntax3` and moved test for `|` in the  `MigrateOptionSyntax`.
Also added tests and migration for the `|` method of `\/`

gabro reviewed (approved):
LGTM! Thanks for this contribution :)
